### DISCUSSION TITLE
useForNotification - handle that only one at a time

### DISCUSSION THREAD

**Discussion by alexbainbridge** [mediagrid/capacitor-native-audio#20] [2. Feature Ideas]
Could you handle useForNotification state, and if one is set (and one is already set) that the prior (existing) one is destroyed first

We have multiple features vying to play audio and managing state at our end is ok, but also a bit of overhead could quite like not to have.... 

Thanks!

**Comment by wsamoht** [comment]
I see your pain point. Are you trying do a playlist or you just have multiple features playing audio one after another?

Are you using the `changeAudioSource` and `changeMetadata` methods? Those were created to easily change the audio for an already added audio item.

**Comment by alexbainbridge** [comment]
Some of our audio is AI generated, so takes e.g. 10 seconds to generate (at point of initialize), so we cant do that preload at the same time we are playing something else.....  beginning to think all our audio should be background, and we just initialize the usefornotificaiton with e.g 1 second of silence, and never actually play or destroy it.....  

No, we are destroying and restarting everytime, mostly because it kind of "just works" ;)

**Reply by wsamoht** [reply]
Got it. It sounds like more of a playlist style would be better. Several people are using this plugin for a playlist which was not thought of when I created it as we didn't need it at for our app. But, I am considering doing some rewrite to support a playlist and changing how background music is added.

**Comment by robinrobinrobinrobin** [comment]
I would love to have a functionality like this! I'm working on a podcast app and I'm trying to preload episodes before a user decides to play them. With the current setup of the audio plugin it's impossible for me to preload an audio stream.

**Reply by wsamoht** [reply]
A workaround is possibly downloading/caching the audio and passing a local file path to `initialize` or `changeAudioSource`. A local file path was confirmed over here -  https://github.com/mediagrid/capacitor-native-audio/discussions/27#discussioncomment-13123868. For a podcast app, this seems like the best option anyways especially if you want to support offline mode or download now and play later.

**Reply by robinrobinrobinrobin** [reply]
Hey Tom. Thanks for your reply! That sounds like an interesting idea! I'll be looking into that a bit deeper. On slower internet connections it might be quicker to load the stream than waiting for the entire episode to download.
